src/sections/blog-post.liquid

```diff
@@ -0,0 +1,147 @@
+{%- comment -%}
+  This is a required section for the Shopify Theme Store.
+  It is available when you add "Blog Posts" section on the Theme Editor.
+
+  Theme Store required settings
+  - Blog (which blog category to show in your store)
+  - Number of posts to show
+
+  Theme Store optional settings
+  - Show author
+  - Show date
+{%- endcomment -%}
+<section class="blog-post">
+  {% if section.settings.title != blank %}
+    <h2 class="blog-post__title">{{ section.settings.title | escape }}</h2>
+  {% endif %}
+
+  {%- comment -%}
+    This refer from blog section when you add the section from Theme Editor
+  {%- endcomment -%}
+  {%- assign blog = blogs[section.settings.blog] -%}
+
+  {% unless blog.articles_count == 0 %}
```

Comment:
We don't need this check - the for loop is good enough since it won't iterate if there aren't any blog articles. You'll need to add the `{% else %}` inside of this for loop though, e.g.:

```liquid
{% for article in blog.articles %}
  // output blog articles
{% else %}
  {% for i in (1..section.settings.post_limit) %}
      // output placeholders
  {% endfor %}
{% endfor %}